### PR TITLE
#4186 Check delivery and fail tests if delivery failed

### LINKED ISSUE #4186: Keptn on Keptn: Mark test as failed if a delivery fails
Right now there is no way of us knowing that an evaluation in our Meta Keptn fails, other than manually going into Keptn's Bridge:

image

The delivery is started here:
https://github.com/keptn/keptn/blob/0f24233122a216b6bf597731f8673f345b841e0c/.github/workflows/integration_tests.yml#L984-L1011

We should also verify that the delivery has succeeded, e.g., by taking the shkeptncontext of the delivery and querying for `${STAGE}.delivery.finished` events

# Acceptance Criteria

- [ ] Integration test pipeline fails if delivery of Keptn with Keptn fails
- [ ] Integration test pipeline fails if delivery of jmeter-service with Keptn fails
- [ ] Integration test pipeline fails if delivery of helm-service with Keptn fails

### PR DESCRIPTION
**Pull request by mowies**
**This PR**
* introduces new pipeline steps that check if the delivery of {keptn,jmeter-service,helm-service} went through on the keptn-on-keptn instance.
* all steps have a timeout of 5 minutes and fail if the delivery didn't go through in that timeframe

Fixes #4186

Signed-off-by: Moritz Wiesinger <moritz.wiesinger@dynatrace.com>

### CODE REVIEW

**Comment by mowies** [comment]
This is the new integration test workflow structure

Screenshot 2021-06-21 at 16 21 45

File: .github/workflows/integration_tests.yml [hunk 1]
```diff
@@ -1095,33 +1095,35 @@ jobs:
...
+          - SERVICE_NAME: jmeter-service
+          - SERVICE_NAME: helm-service
+    env:
+      SERVICE_NAME: ${{ matrix.SERVICE_NAME }}
+      META_KEPTN_KEPTN_PROJECT: keptn
+    steps:
+      - name: Install Keptn CLI for META_KEPTN
+        id: install_keptn_cli
+        run: |
+          curl -sL https://get.keptn.sh | KEPTN_VERSION=${{ env.META_KEPTN_VERSION }} bash
```
Comment (christian-kreuzberger-dtx) [inline .github/workflows/integration_tests.yml:1119]: This step is executed for every service in the matrix, and takes some time.

Can we cache the Keptn CLI (with a cache key being the specified `META_KEPTN_VERSION`), and only download it if it's not available? e.g. 
```
- name: Cache Keptn CLI
  uses: actions/cache@v2
  with:
    path: |
      /usr/local/bin/keptn <-- not sure if that's the right path
    key: ${{ env.META_KEPTN_VERSION }}
```

Comment (mowies) [reply .github/workflows/integration_tests.yml]: i added the cache step and also added a condition to not fetch the keptn installer script if there was a cache hit ✅

**Review by christian-kreuzberger-dtx — APPROVED** [review]
LGTM

**Pull request merged**